ethereum-consensus/src/capella/block_processing.rs

```diff
@@ -39,10 +42,49 @@ pub fn process_bls_to_execution_change<
         BYTES_PER_LOGS_BLOOM,
         MAX_EXTRA_DATA_BYTES,
     >,
-    _signed_address_change: &mut SignedBlsToExecutionChange,
-    _context: &Context,
+    signed_address_change: &mut SignedBlsToExecutionChange,
+    context: &Context,
 ) -> Result<()> {
-    unimplemented!()
+    let address_change = &mut signed_address_change.message;
+    let signature = &signed_address_change.signature;
+
+    if address_change.validator_index >= state.validators.len() {
+        return Err(invalid_operation_error(InvalidOperation::BlsToExecutionChange(
+            InvalidBlsToExecutionChange::ValidatorIndexOutOfBounds(address_change.validator_index),
+        )))
+    }
+
+    let validator = &mut state.validators[address_change.validator_index];
+    let withdrawal_credentials = &mut validator.withdrawal_credentials;
```

Comment:
feels a bit risky to have this &mut Validator lying around

```suggestion
    let withdrawal_credentials = &mut state.validators[address_change.validator_index].withdrawal_credentials;
```